sql/sql_select.cc

```diff
@@ -2590,6 +2590,11 @@ bool make_join_readinfo(JOIN *join, uint no_jbuf_after) {
               join->thd->inc_status_select_scan();
             else
               join->thd->inc_status_select_full_join();
+            /* Block full table/index scans, if optimizer_full_scan is off. */
```

Comment:
Should be moved above the status var bumps, or they will be bumped for full table scans that did not actually happen

Reply:
It's done